apps/stripe/src/app/api/stripe/webhook/stripe-event-handlers/payment-intent-amount-capturable-updated-handler.ts

```diff
@@ -0,0 +1,84 @@
+import { err, ok, Result } from "neverthrow";
+import Stripe from "stripe";
+
+import { BaseError } from "@/lib/errors";
+import { SaleorMoney } from "@/modules/saleor/saleor-money";
+import { createDateFromStripeEvent } from "@/modules/stripe/stripe-event-date";
+import { StripePaymentIntentId } from "@/modules/stripe/stripe-payment-intent-id";
+import {
+  createStripePaymentIntentStatus,
+  StripePaymentIntentStatusValidationError,
+} from "@/modules/stripe/stripe-payment-intent-status";
+import { AuthorizationSuccessResult } from "@/modules/transaction-result/success-result";
+import { RecordedTransaction } from "@/modules/transactions-recording/transaction-recorder";
+
+import { TransactionEventReportVariablesResolver } from "../transaction-event-report-variables-resolver";
+
+type PossibleErrors = InstanceType<
+  | typeof SaleorMoney.ValidationError
+  | typeof StripePaymentIntentStatusValidationError
+  | typeof PaymentIntentAmountCapturableUpdatedHandler.NotSupportedStatusError
+>;
+
+export class PaymentIntentAmountCapturableUpdatedHandler {
+  static NotSupportedStatusError = BaseError.subclass("NotSupportedStatusError", {
+    props: {
+      _internalName: "PaymentIntentAmountCapturableUpdatedHandler.NotSupportedStatusError" as const,
+    },
+  });
+
+  async processEvent({
+    event,
+    stripePaymentIntentId,
+    recordedTransaction,
+  }: {
+    event: Stripe.PaymentIntentAmountCapturableUpdatedEvent;
+    stripePaymentIntentId: StripePaymentIntentId;
+    recordedTransaction: RecordedTransaction;
+  }): Promise<Result<TransactionEventReportVariablesResolver, PossibleErrors>> {
+    const intentObject = event.data.object;
+    const currency = intentObject.currency;
+    const authorizedAmount = intentObject.amount_capturable;
+    const eventDate = createDateFromStripeEvent(event);
+
+    const paramsResult = Result.combine([
+      SaleorMoney.createFromStripe({
+        amount: authorizedAmount,
+        currency,
+      }),
+      createStripePaymentIntentStatus(intentObject.status),
+    ]);
+
+    if (paramsResult.isErr()) {
+      return err(paramsResult.error);
+    }
+
+    const [saleorMoney, paymentIntentStatus] = paramsResult.value;
+
+    if (paymentIntentStatus !== "requires_capture") {
+      return err(
+        new PaymentIntentAmountCapturableUpdatedHandler.NotSupportedStatusError(
+          "PaymentIntent status is not 'requires_capture'",
+          {
+            props: {
+              status: paymentIntentStatus,
+            },
+          },
+        ),
+      );
+    }
```

Comment:
suggestion: I'm wondering why we accept only `requires_capture` if this exception pops-up, we might end-up guessing later why does it happen - could we maybe add some comment here that explains this?

Reply:
About why only `requires_capture`:
This is assumption that app is making - [`payment_intent.amount_capturable_updated`](https://docs.stripe.com/api/events/types#event_types-payment_intent.amount_capturable_updated) event should have only one status - `requires_capture` as it will be triggered when payment is in manual mode (a.k.a authorization flow). 

I will add logger in the next PR so we will know if there will be other status than `requires_capture` and we have to revisit this assumpiton